### ISSUE TITLE
Feature: Possibility to enable/disable caching for logged in users

### ISSUE THREAD

**Issue by niladam** [litespeedtech/lscache-laravel#7]
It would be great if i could be able to disable cache on an per-auth setting.

I'm thinking this would involve some sort of config key (maybe guests_only) and probably a check if the user is logged in or not.

Also, maybe it could automatically build a private cache (per user setting?) automatically if the user is logged in..

**Comment by lucasRolff** [comment]
Hi @niladam ,

> I'm thinking this would involve some sort of config key (maybe guests_only) and probably a check if the user is logged in or not.

I've pushed this to master, so you can try out dev-master via composer, you'll have to publish the new config:

```
php artisan vendor:publish --provider="Litespeed\LSCache\LSCacheServiceProvider"
```

Then you can either use `LSCACHE_GUEST_ONLY` in your .env file and set it to `true` or set it to `true` directly in `config/lscache.php`.

Let me know if it works for you, then I'll get it tagged and release as an official version.

> Also, maybe it could automatically build a private cache (per user setting?) automatically if the user is logged in..

I like the idea, but it's a bit harder, since we have to make purging work as expected as well (purging both public and private cache when using URIs, tags etc), but I'll see what I can do!

**Comment by niladam** [comment]
Hey @lucasRolff - thank you for your really quick improvement and addition.

I've made some minor changes to your previous commit and made a pull request about it. I've documented the `guest_only` functionality as well, and also made sure to reference this issue and your initial commit in it. 

When the PR (if) is accepted this should automatically close this issue (as far as i'm aware)

In case you're seeing this first (as opposed to seing the PR) the PR is #8 :)

Thanks,
Madalin.
